torch/_inductor/compile_fx.py

```diff
@@ -174,11 +174,11 @@ def compile_fx_inner(
 
 
 def clone_preserve_strides(x):
-    needed_size = (
-        sum((shape - 1) * stride for shape, stride in zip(x.size(), x.stride())) + 1
-    )
-    buffer = torch.as_strided(x, (needed_size,), (1,)).clone()
-    return torch.as_strided(buffer, x.size(), x.stride())
+    # This is *bad*, x.storage() is deprecated and emits warnings
+    # TODO: figure out a way to get "# elements in the storage, in the current dtype of x"
+    needed_size = x.storage().size()
```

Comment:
If you don't care about any data after the tensor in the storage, you could use `compute_required_storage_length`
```suggestion
    from torch._prims_common import compute_required_storage_length
    needed_size = compute_required_storage_length(x.size(), x.stride(), x.storage_offset())
```

Reply:
Cc @kurtamohler but it should be pretty easy; storage APIs are always in terms of bytes

Reply:
thanks @peterbell10, that's exactly what I was looking for!